tests/test_pipeline.py

```diff
@@ -108,6 +109,76 @@ def test_predict_all_with_novelties(self):
         self.assertEqual(self.testing_mapped_triples.shape[0], all_df['in_testing'].sum())
 
 
+class TestPipelineCheckpoints(unittest.TestCase):
+    """Test the pipeline with checkpoints."""
+
+    def setUp(self) -> None:
+        """Set up a shared result as standard to compare to."""
+        self.random_seed = 123
+        self.model = 'TransE'
+        self.dataset = 'nations'
+        self.checkpoint_name = "PyKEEN_training_loop_test_checkpoint.pt"
+        self.temporary_directory = tempfile.TemporaryDirectory()
+
+    def tearDown(self) -> None:
```

Comment:
@lvermue note I changed the temporary directory usage here to use the test case's tearDown functionality

Reply:
Why that? 👀 
Now there will be orphans when the unit tests are cancelled.

Reply:
no it's smart and runs teardown even if they're cancelled